### PR TITLE
Typescript Fixes

### PR DESCRIPTION
**Pull request by minerj101**

### CODE REVIEW

File: src/classes/pinger.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 export default class Pinger {
-	#pinger?: ReturnType<typeof setTimeout>;
+	#pinger?: NodeJS.Timer | number;
```
Comment (oberbeck) [inline src/classes/pinger.ts:2]: we should be environment generic as we support not only NodeJS but also Deno and so on...

Comment (mathe42) [reply src/classes/pinger.ts:2]: Yes should be ReturnType<...>

**Comment by mathe42** [comment]
Best setup:
1. Use VSCode
2. Install Deno-Extension
3. Install Deno

=> All errors are gone

**Comment by mathe42** [comment]
And yes we need the `.ts` extensions in the files! This repo will be pushed with no change to `deno.land/x/surrealdb`

**Comment by mathe42** [comment]
@minerj101 Thanks for the PR!

The only change that doesn't break anything is the tsconfig. But I am not a big fan of that much comments. So I think this will not get merged.

Could you close this PR?

**Comment by mathe42** [comment]
@minerj101 @tobiemh can be closed

**Pull request closed without merging**